test/evaluate/test_folding.sh

```diff
@@ -104,3 +95,5 @@ else
   echo all $passed tests passed
   echo PASS
 fi
+
+[[ $KEEP ]] || trap "rm -rf $temp" EXIT
```

Comment:
By moving this down here it means the temp directory won't get deleted in case of an early exit. That was the whole point of using trap.

Reply:
I was having an issue getting this to work, as it was complaining about the directory already existing, directly after the `rm -rf`, as well as the logs not existing. I suppose moving this here was part of me debugging it. I'll move it back and see if it still works, I imagine so.

Reply:
A better way to handle the temp directory is to have the run command pass in `%t` and use that. Then presumably lit takes care of cleaning up so the scripts don't have to worry about it.